src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -385,6 +384,17 @@ internal void SerializeEncoding(SerializationInfo info, StreamingContext context
             }
         }
 
+        // On Desktop, encoding instances that aren't cached in a static field are cached in
+        // a hash table by codepage.
+        private static volatile Hashtable encodings;
+#else
+        // On CoreCLR, the only instance that would need to be cached in the hash table is UTF32BE.
+        // Instead of using a hash table, simply cache the instance in a static field.
+        private static volatile Encoding s_utf32BE;
+        private static Encoding UTF32BE =>
```

Comment:
I am wondering whether it would look better to cache this on UTF32Encoding (Unicode encoding is caching two instances as well - one extra tiny potentially unused singleton if you use UTF32Encoding is no big deal).